op-chain-ops/genesis/config.go

```diff
@@ -198,6 +197,12 @@ type DeployConfig struct {
 	// FundDevAccounts configures whether or not to fund the dev accounts. Should only be used
 	// during devnet deployments.
 	FundDevAccounts bool `json:"fundDevAccounts"`
+	// RequiredProtocolVersion indicates the protocol version that
+	// nodes are required to adopt, to stay in sync with the network.
+	RequiredProtocolVersion common.Hash `json:"requiredProtocolVersion"`
+	// RequiredProtocolVersion indicates the protocol version that
+	// nodes are recommended to adopt, to stay in sync with the network.
+	RecommendedProtocolVersion common.Hash `json:"recommendedProtocolVersion"`
```

Comment:
We should probably add something into the `Check` method. If we return an error when its the empty value, then it will break `pnpm validate-deploy-configs` in `contracts-bedrock` and we will need to update all deploy configs with values. Another option is to log a warning if it is empty, which seems sane to do in the short term. Then when its in production, we can error if its not set

Reply:
There is no clear spec how to add optional backwards compatible deploy config values. I think the current approach is correct. I'll open an issue to track it, feel free to tweak the deploy config in a follow-up PR.

Reply:
Tracking in CLI-4401